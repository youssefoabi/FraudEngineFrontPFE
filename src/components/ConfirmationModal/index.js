import { func, node } from 'prop-types';
import { Modal } from '@cdiscount/backoffice-ui';
import React from 'react';

export default function ConfirmationModal({ children, confirm, toggle, ...props }) {
  function handleConfirm() {
    toggle();
    confirm();
  }

  return (
    <Modal
      actions={[
        {
          color: 'default',
          label: 'Non, Annuler',
          onClick: toggle,
          variant: 'contained',
          size: 'large',
        },
        {
          color: 'quaternary',
          label: 'Oui, Supprimer',
          onClick: handleConfirm,
          size: 'large',
          variant: 'contained',
        },
      ]}
      container={document.getElementById('root')}
      toggle={toggle}
      {...props}
    >
      {children}
    </Modal>
  );
}

ConfirmationModal.propTypes = {
  children: node,
  confirm: func,
  toggle: func,
};

ConfirmationModal.defaultProps = {
  children: null,
  confirm: Function.prototype,
  toggle: Function.prototype,
};
