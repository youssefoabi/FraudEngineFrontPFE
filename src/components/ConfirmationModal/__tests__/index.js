import { itRendersCorrectly } from '@cdiscount/jest-utils';
import React from 'react';

import ConfirmationModal from '../';

describe(ConfirmationModal.name, () => {
  itRendersCorrectly(<ConfirmationModal />);
});
