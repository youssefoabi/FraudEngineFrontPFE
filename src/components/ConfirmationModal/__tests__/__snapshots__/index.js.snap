// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal renders correctly 1`] = `
<y
  actions={
    Array [
      Object {
        "color": "default",
        "label": "Non, Annuler",
        "onClick": [Function],
        "size": "large",
        "variant": "contained",
      },
      Object {
        "color": "quaternary",
        "label": "Oui, Supprimer",
        "onClick": [Function],
        "size": "large",
        "variant": "contained",
      },
    ]
  }
  container={null}
  dialogProps={null}
  fullScreen={false}
  isOpen={false}
  toggle={[Function]}
/>
`;
